lib/tasks/spam_report.rake

```diff
@@ -0,0 +1,68 @@
+namespace :spam do
+
+  desc "Print list of potential spammers"
+  task(:print_possible => :environment) do
+    period = 1.day.ago
+    history_period = 14.day.ago
+    threshold = ArchiveConfig.SPAM_THRESHOLD
+    spam_score = Hash.new
+    spam_works = Hash.new
+    new_works = Work.where("created_at > :week and hidden_by_admin = 'false' ", {:week => period})
+    pseud_list = new_works.collect { |w|  unless  w.pseuds.first.nil? 
+                                                  w.pseuds.first.user_id  
+                                          end }.uniq
+    new = []
+    pseud_list.each { |pseud|
+      new = []
+      ips = Hash.new
+      score = 0
+      begin
+        user=User.find(pseud)
+      rescue ActiveRecord::RecordNotFound => e
+        user = nil
```

Comment:
You could save some lines here by using where instead of find:

    user = User.where(id: user_id).first

That will be nil if the id can't be found.